### PR TITLE
api: add new parse api

### PR DESCRIPTION
**Pull request by Bostigger**

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"github.com/profclems/compozify/api/router"
+	"log"
+	"net/http"
```
Comment (profclems) [inline api/main.go:7]: nit: let's reorganize this into:

```
<standard imports>

<thirdparty imports>

<project imports>
```

```suggestion
	"log"
	"net/http"
	
	"github.com/gorilla/mux"
	
	"github.com/profclems/compozify/api/router"
```

File: api/handler/handler.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	err = p.Parse()
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error parsing command: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	dockerComposeYaml := p.String()
+
+	w.Header().Set("Content-Type", "application/x-yaml")
+	_, err = w.Write([]byte(dockerComposeYaml))
```
Comment (profclems) [inline api/handler/handler.go:48]: Let's return the response in JSON. I want to see something like:

```
{
      "output": "<compose>"
}
```

**Review by profclems — CHANGES_REQUESTED** [review]
This is neat! I left a few comments

File: api/handler/handler.go (hunk 2 above)
Comment (Bostigger) [reply api/handler/handler.go:48]: ohokay.

Comment (profclems) [reply api/handler/handler.go:48]: This is much better. No line breaks or indents will be lost once the user parses the JSON and get the output

File: api/main.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package main
+
+import (
+	"github.com/gorilla/mux"
+	"log"
+	"net/http"
+
+	"github.com/profclems/compozify/api/router"
```
Comment (profclems) [inline api/main.go:8]: nit: same arrangement as mentioned before

**Review by profclems — APPROVED** [review]
LGTM! 

I added one non-blocking comment

**Comment by Aeswolf** [comment]
LGTM!

**Comment by Aeswolf** [comment]
LGTM!

**Pull request merged**